src/server/profiles/account/account.controller.js

```diff
@@ -3,41 +3,73 @@ const { resolveFromVersion } = require('../../utils/resolve.utils');
 const responseUtils = require('../../utils/response.utils');
 const errors = require('../../utils/error.utils');
 
-module.exports.search = function search ({ profile, logger, config, app }) {
-	let { serviceModule: service } = profile;
+/**
+ * @description Construct a resource with base/uscore path
+ */
+let getResourceConstructor = (base) => {
+	let Account = require(resolveFromVersion(base, 'base/Account'));
```

Comment:
Unless you're doing something with account, 

you should change to return require(.....);

Reply:
Later on, some resource might have extensions like the one in Observation, so this is getResourceConstructor in Observation. But we can change to just return the require for now.

let getResourceConstructor = (base, resourceType) => {
	let Observation = require(resolveFromVersion(base, 'uscore/Observation'));
	let Results = require(resolveFromVersion(base, 'uscore/Results'));
	let SmokingStatus = require(resolveFromVersion(base, 'uscore/SmokingStatus'));
	switch (resourceType) {
		case Results.__resourceType:
			return Results;
		case SmokingStatus.__resourceType:
			return SmokingStatus;
		default:
			return Observation;
	}
};

Reply:
Wouldn't they have their own class?

Reply:
Rob might be best to answer this, but I think yes; however, when the client requests a resource, using the base resource (Observation) with some binding parameter for the extension (code=xxx-smokingstatus), we'd still need to return the correct resource type "SmokingStatus")